packages/cardpay-cli/utils.ts

```diff
@@ -3,39 +3,84 @@ import Web3 from 'web3';
 import { AbstractProvider } from 'web3-core';
 import { HttpProvider, getConstant, networkIds, getConstantByNetwork } from '@cardstack/cardpay-sdk';
 import WalletConnectProvider from '@walletconnect/web3-provider';
-import { Options } from 'yargs';
+import { Options, Arguments } from 'yargs';
+/* eslint-disable @typescript-eslint/no-require-imports,@typescript-eslint/no-var-requires */
+const TrezorWalletProvider = require('trezor-cli-wallet-provider');
 
 const BRIDGE = 'https://safe-walletconnect.gnosis.io/';
 
-export async function getWeb3(network: string, mnemonic?: string): Promise<Web3> {
-  if (mnemonic) {
-    return new Web3(
-      new HDWalletProvider({
-        chainId: networkIds[network],
-        mnemonic: {
-          phrase: mnemonic,
+export type ConnectionType = 'mnemonic' | 'trezor' | 'wallet-connect';
+
+interface Web3OptsMnemonic {
+  connectionType: 'mnemonic';
+  mnemonic: string;
+}
+
+interface Web3OptsTrezor {
+  connectionType: 'trezor';
+}
+
+interface Web3OptsWalletConnect {
+  connectionType: 'wallet-connect';
+}
+
+export type Web3Opts = Web3OptsMnemonic | Web3OptsTrezor | Web3OptsWalletConnect;
+
+export function getWeb3Opts(args: Arguments): Web3Opts {
+  switch (args.connectionType as ConnectionType) {
+    case 'wallet-connect':
+      return {
+        connectionType: 'wallet-connect',
+      } as Web3OptsWalletConnect;
+    case 'trezor':
+      return {
+        connectionType: 'trezor',
+      } as Web3OptsTrezor;
+    case 'mnemonic':
+      return {
+        connectionType: 'mnemonic',
+        mnemonic: args.mnemonic,
+      } as Web3OptsMnemonic;
+  }
+}
```

Comment:
@habdelra I needed to add `getWeb3Opts` as a way to parse arguments into the type. Its a utility function.